### PR TITLE
[Mac] Brush Refactoring and Tweaks.

### PR DESCRIPTION
**Pull request by CartBlanche**
- [x] Separates out the Brush Resource classes into separate files, for consistency.
- [x] Uses Localised strings instead of hard-coded ones.
- [x] NSOutlineView now reuses Views instead of creating totally new ones each time.
- [x] Grouped Brush files together under it's own subfolder.
- [x] Now draws the ColourNoBrush Colours

### CODE REVIEW

File: Xamarin.PropertyEditing.Mac/Controls/BrushEditorControl.cs [hunk 1]
```diff
@@ -124,6 +124,8 @@ protected override void UpdateValue ()
 			if (ViewModel.Solid != null) {
 				var title = GetTitle ();
 
+				this.popUpButton.ToolTip = ViewModel.Resource != null ? string.Format ("{0} - {1}: {2}", title, Properties.Resources.Resource, ViewModel?.Resource?.Name) : string.Empty;
```
Comment (ermau) [inline Xamarin.PropertyEditing.Mac/Controls/BrushEditorControl.cs]: This should be something on the property button control itself, not something we have to keep replicating everywhere. As it stands this only shows up for solid brushes. See the windows version: https://github.com/xamarin/Xamarin.PropertyEditing/blob/master/Xamarin.PropertyEditing.Windows/PropertyButton.cs#L137

File: Xamarin.PropertyEditing/ViewModels/BrushPropertyViewModel.cs [hunk 2]
```diff
@@ -172,18 +172,42 @@ private void SetBrushType (CommonBrushType type)
...
+					MaterialDesign.SetToClosest ();
+					break;
+
+				case CommonBrushType.NoBrush:
+					Value = null;
+					break;
+
+				case CommonBrushType.Resource:
+					if (Resource != null) {
+						Value = GetCommonBrushForResource (Resource);
```
Comment (ermau) [inline Xamarin.PropertyEditing/ViewModels/BrushPropertyViewModel.cs]: There's no explanation of why this change was made, nor any test covering  it.

**Comment by ermau** [comment]
Each of those 5 bullet points should have been its own commit. If I look back at this in history, all I'm going to see is "Brush alignment tweaks" which tells me nothing.

**Comment by CartBlanche** [comment]
> Each of those 5 bullet points should have been its own commit. If I look back at this in history, all I'm going to see is "Brush alignment tweaks" which tells me nothing.

Good point. The commit message now reflects all the changes made.  See - https://github.com/xamarin/Xamarin.PropertyEditing/pull/364/commits/f04b2ffd08fd52614858b9232e2f208f39e752a7

File: Xamarin.PropertyEditing/ViewModels/BrushPropertyViewModel.cs (hunk 2 above)
Comment (CartBlanche) [reply Xamarin.PropertyEditing/ViewModels/BrushPropertyViewModel.cs]: This was accidentally left in, while I was tracking down NoBrush issue. I'll remove it.

File: Xamarin.PropertyEditing.Mac/Controls/BrushEditorControl.cs (hunk 1 above)
Comment (CartBlanche) [reply Xamarin.PropertyEditing.Mac/Controls/BrushEditorControl.cs]: Thanks!

**Comment by CartBlanche** [comment]
@ermau let me know if there is anything else I've missed here.
